### PR TITLE
Spring Boot Web and Webflux Helpers/Binding for CloudEvents

### PR DESCRIPTION
**Pull request by salaboy**
Here are some helpers for Spring Boot (web and webflux) to parse and send Cloud Events from Spring Boot applications. 
This is based on this blog post which contains some examples (examples in the blog post needs to be updated to use the new version of the helpers provided in this PR) 

https://salaboy.com/2020/02/20/getting-started-with-knative-2020/

### CODE REVIEW

**Comment by salaboy** [comment]
@spencergibb @cmoulliard @olegz @slinkydeveloper I am happy to accept feedback to make this better..

**Comment by slinkydeveloper** [comment]
Hi @salaboy! Thanks for looking at this! I wonder, can you use the message reader/writer APIs like in https://github.com/cloudevents/sdk-java/tree/master/http?

**Comment by salaboy** [comment]
@slinkydeveloper sure I can..

**Comment by slinkydeveloper** [comment]
@salaboy exactly. You need to think at these `Reader`/`Writer` as kind of builders, that allows you to read write cloudevents without caring about the differences between spec versions and "modes". If you need any help with these, feel free to reach me on cncf slack at @slinkydeveloper

File: http/spring-boot-starter-web-cloudevents/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    <version>2.0.0-milestone1</version>
+</dependency>
+```
+
+## Receiving CloudEvents
+Below is a sample on how to read and write CloudEvents from a Post Request Headers and Body:
+
+```java
+    @PostMapping
+    public String recieveCloudEvent(@RequestHeader Map<String, String> headers, @RequestBody Object body) {
```
Comment (metacosm) [inline http/spring-boot-starter-web-cloudevents/README.md:18]: Typo: `recieve` -> `receive` 😄

File: http/spring-boot-starter-web-cloudevents/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# HTTP Protocol Binding for Spring Boot Web (Reactive Stack)
+
+For Maven based projects, use the following to configure the CloudEvents Spring Boot Binding:
+
+```xml
+<dependency>
+    <groupId>io.cloudevents</groupId>
+    <artifactId>spring-boot-starter-web-cloudevents</artifactId>
```
Comment (cmoulliard) [inline http/spring-boot-starter-web-cloudevents/README.md:8]: Who is gonna to release/maintain such a starter like also the WebFlux one ? @salaboy

File: http/spring-boot-starter-web-cloudevents/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+
+## Sending CloudEvents
+
+
+Below is a sample on how to use the client to send a CloudEvent:
+
+```java
+  // Create the CloudEvent with the builder
+  final CloudEvent myCloudEvent = CloudEventBuilder.v03()
```
Comment (cmoulliard) [inline http/spring-boot-starter-web-cloudevents/README.md:34]: Hiding the version  of the spec or the version of the builder (v03, V1, ....) is perhaps not a great idea. I would prefer that this version is part of the GAV consumed by the starter to make it available publicly to all the users.

**Comment by slinkydeveloper** [comment]
@salaboy check out this one https://github.com/cloudevents/sdk-java/pull/288

**Comment by salaboy** [comment]
@slinkydeveloper I will be happy to check if the new PR is covering what I needed to do from my side..

**Comment by salaboy** [comment]
@slinkydeveloper I will close this PR because this was solved already! I am super happy with the results! :)

**Pull request closed without merging**